lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -1456,6 +1507,44 @@ def test_success(self, parent_id, mock_emit):
         self.assertEqual(actual_event_name, expected_event_name)
         self.assertEqual(actual_event_data, expected_event_data)
 
+    @ddt.data(
+        *itertools.product(
+            [
+                FORUM_ROLE_ADMINISTRATOR,
+                FORUM_ROLE_MODERATOR,
+                FORUM_ROLE_COMMUNITY_TA,
+                FORUM_ROLE_STUDENT,
+            ],
+            [True, False],
+            ["question", "discussion"],
+        )
+    )
+    @ddt.unpack
+    def test_endorsed(self, role_name, is_thread_author, thread_type):
+        role = Role.objects.create(name=role_name, course_id=self.course.id)
+        role.users = [self.user]
+        self.register_get_thread_response(
+            make_minimal_cs_thread({
+                "id": "test_thread",
+                "course_id": unicode(self.course.id),
+                "thread_type": thread_type,
+                "user_id": str(self.user.id) if is_thread_author else str(self.user.id + 1),
+            })
+        )
+        self.register_post_comment_response({}, "test_thread")
+        data = self.minimal_data.copy()
+        data["endorsed"] = True
+        expected_error = (
+            role_name == FORUM_ROLE_STUDENT and
+            (not is_thread_author or thread_type == "discussion")
+        )
+        try:
+            create_comment(self.request, data)
+            self.assertEqual(httpretty.last_request().parsed_body["endorsed"], ["True"])
+            self.assertFalse(expected_error)
+        except ValidationError:
+            self.assertTrue(expected_error)
+
```

Comment:
this doesn't deal with groups at all; was it added here to fill a gap discovered while working on this PR?